testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -166,7 +164,7 @@ def test_subselection_alignto(self, universe, reference, subselection, expectati
 
         with expectation:
             rmsd = align.alignto(universe, reference, subselection=subselection)
-            assert_almost_equal(rmsd[1], 0.0, decimal=9)
+            assert_allclose(rmsd[1], 0.0, atol=9)
```

Comment:
`assert_almost_equal` and `assert_allclose` are different functions that take different arguments. You can't just change the name of the function, you also need to modify the (values of the) arguments accordingly.

As an example, try the following:
```
>>> from numpy.testing import assert_almost_equal, assert_allclose
>>> assert_almost_equal(0, 8, 9)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/romeli/miniconda3/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/Users/romeli/miniconda3/lib/python3.9/site-packages/numpy/testing/_private/utils.py", line 604, in assert_almost_equal
    raise AssertionError(_build_err_msg())
AssertionError:
Arrays are not almost equal to 9 decimals
 ACTUAL: 0
 DESIRED: 8
>>> assert_allclose(0, 8, 9)
>>>
```

With the second argument as `9` (now `atol` instead of `decimal`), `assert_allclose` assumes `0` and `8` to be the same.

Please refer to the documentation of `assert_almost_equal` and `assert_allclose` to understand the differences and how to change the arguments.

Reply:
so you are saying that I should change atol=1

Reply:
Not quite. If you look at the [documentation of `assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html), you can see that the _default_ arguments are `rtol=1e-07` and `atol=0`.

You can test what you suggested:
```
>>> assert_allclose(0, 1.1, atol=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/romeli/miniconda3/lib/python3.9/site-packages/numpy/testing/_private/utils.py", line 1592, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/Users/romeli/miniconda3/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/Users/romeli/miniconda3/lib/python3.9/site-packages/numpy/testing/_private/utils.py", line 862, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Not equal to tolerance rtol=1e-07, atol=1

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 1.1
Max relative difference: 1.
 x: array(0)
 y: array(1.1)
>>> assert_allclose(0, 0.9, atol=1)
>>>
```
As you can see, with `atol=1`, `0` and `0.9` are considered to be the same.

What I was suggesting was to carefully read the [documentation of `assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html) (especially the `rtol` and `atol` arguments) and the  [documentation of `assert_almost_equal`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html) (especially the `decimal` argument), to understand what they do and their arguments and the relation between them, so that they can be changed accordingly.

These comments refer to all the changes in this PR, not just the one that I highlighted here.

Reply:
I have made the changes you asked for in these two files. All the checks are successful except the linter error which has already raised as an issue "Try to fix the issue with darker lint bot falling over with PRs from forks #4081"

Reply:
Thanks for your work! I'll give it a proper look in the coming days. In the meantime and to facilitate the review, can you please give your rationale for when you used atol and when you used rtol?

We will need to be very careful when reviewing this PR. Since you are changing the assertions in the tests, the test passing is not necessarily a good indicator. =)

Reply:
I have used atol when the array value is close to zero and I have used rtol when the array value is far from zero.